src/main/java/com/ssu/schedule/parser/ParserSSU.java

```diff
@@ -0,0 +1,208 @@
+package com.ssu.schedule.parser;
+
+import com.ssu.schedule.model.*;
+import com.ssu.schedule.repository.FacultyRepository;
+import com.ssu.schedule.repository.GroupRepository;
+import okhttp3.Credentials;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.XML;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Component
+public class ParserSSU {
+
+    private static final Logger log = LoggerFactory.getLogger(ParserSSU.class);
+
+    @Value("${university.basicAuth.login}")
+    private String login;
+
+    @Value("${university.basicAuth.password}")
+    private String password;
+
+    @Value("${university.url}")
+    private String url;
+
+    @Value("${university.name}")
+    private String univerName;
+
+    @Value("${university.abbr}")
+    private String univerAbbr;
+
+    private final FacultyRepository facultyRepository;
+    private final GroupRepository groupRepository;
+
+    private TimeList timeList = new TimeList();
+
+    @Autowired
+    public ParserSSU(FacultyRepository facultyRepository, GroupRepository groupRepository) {
+        this.facultyRepository = facultyRepository;
+        this.groupRepository = groupRepository;
+    }
+
+    @Scheduled(cron = "0 0 12 * * ?")
+    private void getCurrentSchedule() {
+        OkHttpClient client = new OkHttpClient();
```

Comment:
you don't need to create client and requestBuilder for each request, create two beans instead and inject (@autowire) them whenever they're used